builder/simplify.cc

```diff
@@ -756,120 +786,95 @@ class simplifier : public node_mutator {
       return;
     }
 
-    if (const call* bc = base.as<call>()) {
+    if (const call* bc = is_intrinsic(base, intrinsic::buffer_at)) {
```

Comment:
nit: I know `is_intrinsic` is pre-existing, but it seems like it should be named `as_intrinsic` since it returns an op rather than a bool